### PR TITLE
fix(docs-infra): improve focus styles in topnav and footer

### LINKED ISSUE #33239: Top Menu highlight wrong item after click on browser back button
# 📚 angular.io bug
### Reproduction Steps**
1. Open angular.io page
2. Click on resources link on top
3. Click on docs link on top
4. Click on browser back button.
The docs link is highlighted while it should highlight resource link

### Expected vs Actual Behavior**
Browser back button does change the page back to the last page user on, and it should also the highlight item of current page. But it is not highlighting the correct menu item

### PR DESCRIPTION
**Pull request by sjtrimble**
Fixes #33239

Improved focus styles for topnav and footer including visible color.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Topnav:
focus-topnav-old

Footer:
focus-footer-old

Issue Number: #33239

## What is the new behavior?
Topnav:
focus-topnav-new

Footer:
focus-footer-new

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by petebacondarwin — APPROVED** [review]
Hi @sjtrimble - good to see you back to work.
When I look at the last preview URL, it doesn't look the same as your "new behaviour" animations.
Are they out of date?

Apart from that the focussing looks much better than before.

Also, perhaps out of range for this PR, but it is weird that if the browser is the same width as in your example animations, focussing the search box actually make it go smaller. Can we fix that?

**Comment by sjtrimble** [comment]
@petebacondarwin Heya! Weird, to me the preview URL and the GIF look the same. What is different?

That is super weird about the search box! I'll open an issue for it. It seems to happen in a smaller screen size.

**Comment by sjtrimble** [comment]
@gkalpak here's the PR that needs your help with getting `active` state/class on the topnav links for styling.

**Comment by gkalpak** [comment]
@sjtrimble, let's fix the focus styles on this PR and I'll create a separate one for highlighting active tobbar nav links.

As Pete mentioned, the styling on the preview looks different than your gifs :grin:
Here is what it looks like on my laptop:

focus-styles

The main (undesirable) differences I see are:
- There is no space between the logo and the focus outline.
- The topbar nav links still have the lighter background color (in addition to the outline) when focused.
- There is too much space between the Twitter and GitHub logos and their focus outline (and each outline intersects with the other logo).

Does it look different on your machine?

**Comment by sjtrimble** [comment]
@gkalpak Super strange .. yeah, it doesn't look like that on my machine. I'm using a Chrome browser on my Mac both on my local version and in the PR preview link 🤔

I checked on Safari and it changes the color there for some of the outlines which is also strange ..
Screen Shot 2019-10-22 at 11 10 37 AM
Screen Shot 2019-10-22 at 11 10 22 AM
Screen Shot 2019-10-22 at 11 10 12 AM

**Comment by sonukapoor** [comment]
@gkalpak I have a fix for this ready. I will discuss it with you offline.

**Comment by sonukapoor** [comment]
@sjtrimble / @gkalpak Please see followup PR with a fix:

https://github.com/angular/angular/pull/34634

**Review by gkalpak — APPROVED** [review]
I pushed a fixup commit to fix the outline styling issue mentioned in https://github.com/angular/angular/pull/33255#issuecomment-544885708 and couple more minor issues I saw (see images below).

If anyone is interested, the outline issue was caused by the fact (bug?) that `outline-offset` is not applied on Chrome on Windows, if `outline-style` is set to `auto` 🤷‍♂
(Credits: [StackOverflow question][1])

---
_**Focus outlines before my fix:**_
topbar-outlines before

##
_**Focus outlines after fixing `outline-style`:**_<br />
topbar-outlines after1

##
_**Focus outline of small home icon after fixing `outline-style`:**_<br />
topbar-homeicon-outline-small before

##
_**Focus outlines of home icon after adding some padding:**_<br />
topbar-homeicon-outline after

topbar-homeicon-outline-small after

##
_**External icons before fixing margins:**_<br />
topbar-icon-margins before

_**External icons after fixing margins:**_<br />
topbar-icon-margins after

[1]: https://stackoverflow.com/questions/17141865/outline-offset-does-not-get-applied-on-chrome-windows-when-outline-style-is-aut

**Comment by gkalpak** [comment]
@googlebot I consent.

**Comment by gkalpak** [comment]
@sjtrimble, could you take a look at my fixup commit and verify you are happy with the changes and they look OK on macOS?

**Comment by gkalpak** [comment]
I pushed another fixup commit to fix CI.

**Comment by sonukapoor** [comment]
@gkalpak, it seems that the lighter background is still applied in your version. This was fixed in my PR + I also applied other fixes to the styles. 

image

**Comment by gkalpak** [comment]
> it seems that the lighter background is still applied in your version

Oh, I thought that was intentional. If not, we should get rid of it.

**Comment by sonukapoor** [comment]
@gkalpak. I have fixed my PR again. Thanks

**Comment by gkalpak** [comment]
As discussed "offline", the two PRs are essentially identical, so we will go with this one (as it has more context). Thx for looking into it, @sonukapoor.

File: aio/src/styles/1-layouts/_footer.scss [hunk 1]
```diff
@@ -22,27 +22,39 @@ footer {
...
-      &:visited {
-        text-decoration: none;
-      }
+    }
+
+    &:focus {
+      outline-color: rgba($white, 0.8);
+      outline-offset: 2px;
+      // `outline-offset` is not applied on Chrome on Windows, if `outline-style` is `auto.
+      outline-style: solid;
```
Comment (sjtrimble) [inline aio/src/styles/1-layouts/_footer.scss]: Need to add `outline-width: 1px` so it doesn't look excessive.

Screen Shot 2020-02-18 at 2 57 35 PM

File: aio/src/styles/1-layouts/_top-menu.scss [hunk 2]
```diff
@@ -96,7 +96,18 @@ aio-shell.folder-tutorial mat-toolbar.mat-toolbar {
...
   cursor: pointer;
   margin: 0 16px 0 0;
-  padding: 21px 0;
+  padding: 8px 0;
+
+  &:focus {
+    outline-color: $focus-outline-ondark;
+    outline-offset: 4px;
+    // `outline-offset` is not applied on Chrome on Windows, if `outline-style` is `auto.
+    outline-style: solid;
```
Comment (sjtrimble) [inline aio/src/styles/1-layouts/_top-menu.scss]: Need to add `outline-width: 1px` so it doesn't look excessive.

Screen Shot 2020-02-18 at 3 01 52 PM

File: aio/src/styles/1-layouts/_footer.scss (hunk 1 above)
Comment (gkalpak) [reply aio/src/styles/1-layouts/_footer.scss]: Done.

File: aio/src/styles/1-layouts/_top-menu.scss (hunk 2 above)
Comment (gkalpak) [reply aio/src/styles/1-layouts/_top-menu.scss]: Done (and also for the [social icons][1]).

[1]: https://github.com/angular/angular/blob/02ffecd72/aio/src/styles/1-layouts/_top-menu.scss#L246-L250

**Comment by gkalpak** [comment]
Explicitly set the `outline-width` to 1px. Thx, @sjtrimble!
I am marking this for merging. If there are more tweaks needed, we can make them in a follow-up PR.

**Comment by gkalpak** [comment]
(The payload size failure in `test_aio_local` will be fixed with #35538.)

**Pull request closed without merging**